frontend/src/components/CalendarsToPrint.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {useRef} from "react";
+import {countDaysLeftInWeek} from "../utils/CountDaysLeftInWeek";
+import {Views} from "react-big-calendar";
+import NewCalendar from "./Calendar";
+import CustomWeekView from "../utils/CustomWeekView";
+import ReactToPrint from "react-to-print";
+import "./styles/CalendarsToPrint.css"
+import {EventData} from "../models/Activity";
+
+function CalendarsToPrint({
+                              rangeStart,
+                              rangeEnd,
+                              cbxDeleteDatesChecked,
+                              cbxSelectDateChecked,
+                              roomNumber,
+                              roomActivities,
+                              backgroundEvent,
+                              handleSelectSlot,
+                              onAfterPrint
+                          }: {
+    rangeStart: Date,
+    rangeEnd: Date,
+    cbxDeleteDatesChecked: boolean,
+    cbxSelectDateChecked: boolean,
+    roomNumber?: number,
+    roomActivities: EventData[],
+    backgroundEvent: any,
+    handleSelectSlot: any,
+    onAfterPrint: () => void
+}) {
+    const CalendarsToPrintForwardRef = React.forwardRef((props: any, ref) => {
+        let startDate: Date = rangeStart;
+        let endDate: Date = rangeEnd;
+
+        const formats: any = {}
+        if (cbxDeleteDatesChecked) {
+            formats.dayFormat = (date, culture, localizer) => localizer.format(date, 'EEE', culture);
+            formats.dayRangeHeaderFormat = ({start, end}, culture, localizer) => localizer.format(start, ' ', culture);
+        }
+
+        const datesSelected = cbxSelectDateChecked && startDate != null && endDate != null;
+        const oneWeekTime = 1000 * 60 * 60 * 24 * 7;
+        let calendarsCount = 1;
+        let startDay = startDate.getDay();
+        let endDay = endDate.getDay();
```

Comment:
place it on top of component, above if from 36 line

Reply:
https://stackoverflow.com/a/1411495
Examplary voice in debate where place variables to. It is not good practice to declare all variables on the top of the scope of method.
"It is generally recommended to declare variables as close to their first use as possible, rather than at the top of the scope."

I will adjust my code to your review only bcs rest of code applies this rule, so it will be consistent, but I will say it again - it isn't good practice.
If You want to learn about clean code, I can recommend "Clean Code: A Handbook of Agile Software Craftsmanship" by Robert C. Martin - I am reading it right now and it's really worth reading.